tests/test_regressions.py

```diff
@@ -324,17 +324,46 @@ def test_drf_format_suffix_parameter(no_warnings):
     def pi(request, format=None):
         pass  # pragma: no cover
 
-    urlpatterns = [path('pi', pi)]
-    urlpatterns = format_suffix_patterns(urlpatterns, allowed=['json', 'html'])
+    @extend_schema(responses=OpenApiTypes.FLOAT)
+    @api_view(['GET'])
+    def subpath(request, format=None):
+        pass  # pragma: no cover
+
+    @extend_schema(responses=OpenApiTypes.FLOAT)
+    @api_view(['GET'])
+    def similar(request, format=None):
+        pass  # pragma: no cover
+
+    urlpatterns = [
+        path('pi', pi),
+        path('pi/', pi),
+        path('pi/subpath', subpath),
+        path('pick', similar),
+    ]
+    allowed = ['json', 'html']
+    urlpatterns = format_suffix_patterns(urlpatterns, allowed=allowed)
 
     generator = SchemaGenerator(patterns=urlpatterns)
     schema = generator.get_schema(request=None, public=True)
     validate_schema(schema)
-    assert len(schema['paths']) == 2
+
+    assert len(schema['paths'].keys()) == 7
+    assert list(schema['paths'].keys()) == [
+        '/pi',
+        '/pi/',
+        '/pi{format}',
+        '/pi/subpath',
+        '/pi/subpath{format}',
+        '/pick',
+        '/pick{format}',
+    ]
+
     format_parameter = schema['paths']['/pi{format}']['get']['parameters'][0]
     assert format_parameter['name'] == 'format'
     assert format_parameter['required'] is True
     assert format_parameter['in'] == 'path'
+    assert format_parameter['schema']['type'] == 'string'
+    assert format_parameter['schema']['enum'] == allowed
```

Comment:
this needs to be either

`.. == ['.' + format for format in allowed]`

or the paths need to be rewritten to be `'/pi.{format}'`

If we inject the `.` into the URL, the default alpha sorting isnt quite so bad.

I like the enum being `['.json', ..]` , because then we can also merge paths using `['.json', .., '/']` 

However then we loose the obviousness of the trailing `'/'`.

fwiw, I expect that I could implement this as a post-processor, so if you dont like any of the options, I will do that.

Reply:
The '.' is DRF.

It just hides that detail because it is a non-pretty redundant detail.

c.f. https://github.com/encode/django-rest-framework/blob/bb795674f86828fc5f15d6d61501cc781811e053/rest_framework/urlpatterns.py#L16 and https://github.com/encode/django-rest-framework/blob/335054a5d36b352a58286b303b608b6bf48152f8/rest_framework/schemas/generators.py#L127